### PR TITLE
Routing DSL: style guide

### PR DESCRIPTION
**Pull request by ennru**
Add some guidance how we recommend users organise their routing DSL code.

References #2980

### CODE REVIEW

File: docs/src/main/paradox/routing-dsl/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Routing DSL style guide
+
+Akka HTTP's routing DSL is at the center of most Akka HTTP-based servers. It's where the incoming requests diverge into the different parts of the implemented services.
+
+Keeping all routing in one big structure will easily become hard to grasp and maintain. This page is tries to give a few hints for how you may want to break down the routing logic.
```
Comment (zwift-mark) [inline docs/src/main/paradox/routing-dsl/style-guide.md:5]: `is tries` --> `tries` ?

Comment (ignasi35) [reply docs/src/main/paradox/routing-dsl/style-guide.md:5]: I would even remove the `tries` altogether:

```suggestion
Keeping all routing in one big structure will easily become hard to grasp and maintain. This page gives a few hints for how you may want to break down the routing logic.
```

File: docs/src/test/scala/docs/http/scaladsl/server/directives/StyleGuideExamplesSpec.scala [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+      val customerRoutes: Route =
+        concat(
+          path("customer" / IntNumber) { customerId =>
+            complete("")
+          }
+        // ...
+        )
+
+      val prefer: Route =
```
Comment (ignasi35) [inline docs/src/test/scala/docs/http/scaladsl/server/directives/StyleGuideExamplesSpec.scala:82]: Add a comment here highlightng the composition.

```suggestion
      // 2. Then compose the relative routes under their corresponding path prefix.
      val prefer: Route =
```

File: docs/src/test/scala/docs/http/scaladsl/server/directives/StyleGuideExamplesSpec.scala [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+          }
+        }
+      )
+      // #path-prefix
+    }
+
+    "split" in {
+      // #path-compose
+      // prefer
+      val itemRoutes: Route =
```
Comment (ignasi35) [inline docs/src/test/scala/docs/http/scaladsl/server/directives/StyleGuideExamplesSpec.scala:60]: Add a comment here that first we cerate a couple of partial, relative path matchers:

```suggestion
    // 1. First, create partial matchers (with a relative path)
    val itemRoutes: Route =
```

**Comment by ennru** [comment]
Thank you for hopping on this. Do you think it is useful in this form? Does the structure make sense?

**Comment by ennru** [comment]
Thank you for the feedback. I'll work more on this to find a better scope.
The least I need to do is to complement this PR with Java examples.

File: docs/src/test/scala/docs/http/scaladsl/server/directives/StyleGuideExamplesSpec.scala [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+          path("show" / Segment) { itemId =>
+            get {
+              complete("")
+            }
+          }
+        )
+
+      val customerRoutes: Route =
+        concat(
+          path("customer" / IntNumber) { customerId =>
```
Comment (mfulgo) [inline docs/src/test/scala/docs/http/scaladsl/server/directives/StyleGuideExamplesSpec.scala:78]: This should have been
```scala
path(IntNumber) { customerId =>
```
to match the Java and avoid a confusing `/customer/customer/123` route.

Comment (ennru) [reply docs/src/test/scala/docs/http/scaladsl/server/directives/StyleGuideExamplesSpec.scala:78]: Thank you. By coincidence we got a fix for this the other day in https://github.com/akka/akka-http/pull/3448

Comment (mfulgo) [reply docs/src/test/scala/docs/http/scaladsl/server/directives/StyleGuideExamplesSpec.scala:78]: Perfect! Thanks.

**Pull request merged**